models/netsuite2/netsuite2__transaction_details.sql

```diff
@@ -101,19 +101,20 @@ transaction_details as (
     accounting_periods.is_closed as is_accounting_period_closed,
     accounts.name as account_name,
     accounts.type_name as account_type_name,
+    accounts.account_type_id,
     accounts.account_id as account_id,
     accounts.account_number
 
     --The below script allows for accounts table pass through columns.
     {{ fivetran_utils.persist_pass_through_columns('accounts_pass_through_columns', identifier='accounts') }},
 
     accounts.is_leftside as is_account_leftside,
-    lower(accounts.type_name) like 'accounts payable%' as is_accounts_payable,
-    lower(accounts.type_name) like 'accounts receivable%' as is_accounts_receivable,
+    lower(accounts.account_type_id) = 'acctpay' as is_accounts_payable,
+    lower(accounts.account_type_id) = 'acctrec' as is_accounts_receivable,
     lower(accounts.name) like '%intercompany%' as is_account_intercompany,
```

Comment:
It looks like we didn't update this account name. I agree with you that it seems this is the correct move as there is not a lookup key for these fields. But would you be able to just add some clarification in this thread for posterity.

Reply:
This is a good point. I hadn't really considered it, but I looked into it, and there is a flag in a column called `eliminate` in the accounts source table that is meant to flag an account as intercompany. See the [netsuite docs ](https://www.netsuite.com/help/helpcenter/en_US/srbrowser/Browser2016_1/schema/record/account.html) for field `eliminate`. See also [this doc](https://docs.oracle.com/en/cloud/saas/netsuite/ns-online-help/section_N1475891.html#Making-Intercompany-Journal-Entries) (and search for "eliminate").

I added this to the source and tried this out, however unlike my other changes, this does produce different results from our current model. For example our data has some accounts marked as `eliminate = T`, but the account name does not contain `intercompany`. Thinking this needs to be a new issue for further investigation?

Reply:
Thanks for investigating! This is interesting, however I would prefer we do not make the update here as you can see internally that the changes without the eliminate update tie out, but with the eliminate update they do not. I don't feel there is a need for a new issue as there currently is not an existing issue around this.

Let's discuss at standup what our internally procedures may to flag something like this. I don't feel it is necessary to be an issue, but we also don't have enough information to create a FR.

Reply:
From discussion, we will leave this flag as-is for now, but will come back to it.